easybuild/easyblocks/a/adf.py

```diff
@@ -0,0 +1,95 @@
+##
+# Copyright 2016-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing ADF, implemented as an easyblock
+"""
+import os
+import shutil
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_ADF(EasyBlock):
+    """Support for building/installing ADF."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for ADF."""
+        super(EB_ADF, self).__init__(*args, **kwargs)
+
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'adf<version>' part to avoid having everything in a subdirectory
+        self.cfg['unpack_options'] = "--strip-components=1"
+        super(EB_ADF, self).extract_step()
+
+    def configure_step(self):
+        """Custom configuration procedure for ADF."""
+
+        env.setvar('ADFHOME', self.installdir)
+        env.setvar('ADFBIN', os.path.join(self.installdir, 'bin'))
+        env.setvar('ADFRESOURCES', os.path.join(self.installdir, 'atomicdata'))
```

Comment:
ah, seems like you need to unpack both the sources and binary version, and then recompile

http://www.scm.com/Doc/Doc2014/Install/Installation/page5.html

that explains why some stuff is missing